src/core/bpf/socket_bind/socket-bind-api.bpf.h

```diff
@@ -46,6 +46,7 @@ struct socket_bind_rule {
         __u32 protocol;
         __u16 nr_ports;
         __u16 port_min;
+        _Bool initialized;
```

Comment:
any particular reason why you use `_Bool` rather than `bool` here?

Reply:
hmm, so changing the size of the structure sucks for two reasons: we cannot change the structure for running systems really, i.e. "systemctl daemon-reexec" just breaks here. And there might be a ton of entries in some setups, hence increasing it in size should maybe not be done willy-nilly...

wouldn't it be a better approach to just always insert at least one entry? i.e. adding the right one if none is configured?

Reply:
> any particular reason why you use `_Bool` rather than `bool` here?

I had some wrong assumptions about the other types in the struct. I'm not too familiar with C and sometimes it shows when experience with other languages isn't enough :sweat_smile:

> hmm, so changing the size of the structure sucks for two reasons: we cannot change the structure for running systems really, i.e. "systemctl daemon-reexec" just breaks here. And there might be a ton of entries in some setups, hence increasing it in size should maybe not be done willy-nilly...

I see

> wouldn't it be a better approach to just always insert at least one entry? i.e. adding the right one if none is configured?

I initially tried just inserting null values which didn't work (correct me if there's a way, that would make things a lot easier)

As far as I can see there is (with the current matching) no way to express "match no address".

As the matching code currently ignores anything but the 8 least most significant bits from the address family passed to the syscall (and anything not `AF_INET` or `AF_INET6` for either `user_family` or `family`) anyways we could put a constant in the 32 bit unsigned rule indicating this.